persist/fs/commitlog/commit_log.go

```diff
@@ -99,14 +99,16 @@ type commitLogWrite struct {
 	datapoint    ts.Datapoint
 	unit         xtime.Unit
 	annotation   ts.Annotation
+	ctx          context.Context
```

Comment:
Nit: maybe have this as the first field in the struct?  There's a linter rule that got recently added that complains about ctx not being passed as the first arg in methods, may as well put it at the top of structs too in case that is another rule that gets added...